internal/controller/cloud-control/iprange_azure_test.go

```diff
@@ -205,6 +207,36 @@ var _ = Describe("Feature: KCP IpRange for Azure", func() {
 			Expect(azureSecurityGroup).NotTo(BeNil())
 		})
 
+		var virtualNetworkLink *armprivatedns.VirtualNetworkLink
+		resourceGroupName := azurecommon.AzureCloudManagerResourceGroupName(scope.Spec.Scope.Azure.VpcNetwork)
+		privateDnsZoneName := azureutil.NewPrivateDnsZoneName(kcpIpRangeName)
+		By("Then Azure Virtual Network Link is created", func() {
+			Eventually(func() error {
+				vnl, err := azureMock.GetVirtualNetworkLink(infra.Ctx(), resourceGroupName, privateDnsZoneName, kcpIpRangeName)
+				if err != nil {
+					return fmt.Errorf("error loading azure Virtual Network Link: %w", err)
+				}
+				virtualNetworkLink = vnl
+				return nil
+			}).
+				Should(Succeed())
+			Expect(virtualNetworkLink).NotTo(BeNil())
+		})
+
+		var privateDnsZone *armprivatedns.PrivateZone
+		By("Then Azure Private Dns Zone is created", func() {
+			Eventually(func() error {
+				pdz, err := azureMock.GetPrivateDnsZone(infra.Ctx(), resourceGroupName, privateDnsZoneName)
+				if err != nil {
+					return fmt.Errorf("error loading azure Private Dns Zone: %w", err)
+				}
+				privateDnsZone = pdz
+				return nil
+			}).
+				Should(Succeed())
+			Expect(privateDnsZone).NotTo(BeNil())
+		})
```

Comment:
Same as comment above

Reply:
No, in the case of Azure we know the reference to dnsZone and virtualPrivateLink based on "well known name" pattern (UUID).

We do not add this ref to KCP CR (as we do for the AWS or CCEE), so I can not use proposed pattern.